agent/catalog_endpoint.go

```diff
@@ -201,9 +201,11 @@ func (s *HTTPServer) CatalogServiceNodes(resp http.ResponseWriter, req *http.Req
 	if out.ServiceNodes == nil {
 		out.ServiceNodes = make(structs.ServiceNodes, 0)
 	}
-	for _, s := range out.ServiceNodes {
+	for i, s := range out.ServiceNodes {
 		if s.ServiceTags == nil {
-			s.ServiceTags = make([]string, 0)
+			clone := *s
+			clone.ServiceTags = make([]string, 0)
+			out.ServiceNodes[i] = &clone
```

Comment:
One more spot https://github.com/hashicorp/consul/pull/3867/files#diff-d1304019b26a1b30439750957b23fe06R152

Reply:
and https://github.com/hashicorp/consul/blob/1c6de1d623983868d60078163d351746e75a3311/agent/catalog_endpoint.go#L121